### DISCUSSION TITLE
Button labelling to make draft publish workflow clearer

### DISCUSSION THREAD

**Discussion by simonrjones** [craftcms/cms#13674] [Ideas]
**Problem**
One of our clients have got a bit confused over the labels used to publish and save draft entries in Craft. This has resulted in news articles being accidentally published live before they were intended to be. It's happened a few times now and when I first used Craft (for our staff handbook) I made the same mistake. 

I wondered if the button labelling could be improved to make it clearer (without the user having to have specialist knowledge).

**How to replicate**
1. Create a new entry
2. Select "Save draft" to save as a draft
3. Go back to entry and make an edit
4. At this point the buttons to save the document are ambiguous unless you know what's going on. If you select Create entry thinking it will save the post but not publish it, it actually publishes it live

These buttons saves any changes and publish the draft live: Create entry, Create and continue editing, Create and add another

This button saves any changes as draft: Save draft

**Suggestions**

In the same scenario (in Classic editor) WordPress uses the same language "Save draft" to create a draft, but more clearly has a button called "Publish" to make this live. It also displays the page status next to these buttons which makes this clearer.

I'd suggest:
* Change the word "Create" to "Publish" when an action changes the entry to live status
* Move the status higher up the page so it's easier to spot this and associate the different save state buttons with the current entries status

Alternatively if the Draft & Publish functionality is considered important enough you could have separate buttons for this to make it super clear to users rather than combine this in one button / menu.

**Comment by brandonkelly** [comment]
The actual “live” state of the entry is going to be determined by its status (enabled or disabled), and Post/Expiry Dates, which is why we’ve avoided using the word “Publish”.

If the entry’s status is set to disabled, it’s not obvious what a “Publish” button would do — save it as-is (misnomer), or save it as enabled (ignoring the current status setting)?

Perhaps we could consider automatically changing the label depending on the status. Not exactly sure what the ideal button label would be for an enabled entry with a future Post Date, though. Or an entry that is enabled in one site but not another…

**Reply by brandonkelly** [reply]
One thing you could do to prevent this from ever accidentally happening is change the section’s Default Status to disabled (in the section’s settings). So each new entry would begin life disabled, and need to be explicitly enabled before it will get published live.

**Comment by simonrjones** [comment]
That is really interesting, that does explain how things work better. I appreciate this is a fairly complex area with lots of possibilities.

When you're using the editor for day-to-day work it takes a bit of thinking about to remember the rules you mention above. I think anything to make it clearer the current publication status and when the entry will publish live would be valuable. 

It's hard to know the best solution without testing. A few options come to mind.

1) Amend button labels based on status and post/expire dates. If saving:
- makes an entry live "Publish now" 
- makes an entry in the future "Publish in future"
- doesn't change status "Save" (as now)

2) This could also be clarified by adding more information about the publication status on entry pages. For example, next to the ID, Status, etc, information bottom-right you could add "Publish status" above "Status" to make it clear the current status of a page for the user with values such as:

- This entry is live
- This entry is draft and is not live
- This entry will publish live immediately
- This entry will publish live on DATE
- This entry is live and will go offline on DATE
- This entry is disabled and is not live

Ideally bringing this information higher up the page would be useful.

**Comment by BenParizek** [comment]
We have authors on our team that also find themselves uncertain on whether an entry is live or not when saving an entry. I think Simon has proposed some interesting ideas and would echo that having some method that summarized all these different details (Draft, Post/Expiry, Enabled) for the average user while they were editing, so they were certain what was going to take place when they hit save, would be helpful.
